pkg/store/prometheus_test.go

```diff
@@ -430,6 +430,99 @@ func TestPrometheusStore_Series_MatchExternalLabel(t *testing.T) {
 	testutil.Equals(t, 0, len(srv.SeriesSet))
 }
 
+func TestPrometheusStore_Series_LimitMaxMatchedSeries(t *testing.T) {
+	defer testutil.TolerantVerifyLeak(t)
+
+	p, err := e2eutil.NewPrometheus()
```

Comment:
e2e tests are usually under `test/e2e/query_test.go`. In fact, I didn't even notice that this was an e2e test until I ran this and it failed for me :smile:

Reply:
It's already a mess, nvm